app/rtc_compare.py

```diff
@@ -0,0 +1,483 @@
+        flag_discrepancy = (val_1 != val_2)
+        index_first_discrepancy = np.where(flag_discrepancy)[0][0]
+        print(f'{str_indent} The first discrepancy has detected from index '
+              f'[{index_first_discrepancy}] : '
+              f'1st=({val_1[index_first_discrepancy]}), '
+              f'2nd=({val_2[index_first_discrepancy]})')
+
+    # Check pixel-by-pixel nan / non-nan difference
+    if (issubclass(val_1.dtype.type, np.floating) or
+        issubclass(val_1.dtype.type, np.complexfloating)):
+
+        mask_nan_val_1 = np.isnan(val_1)
+        mask_nan_val_2 = np.isnan(val_2)
+
+        mask_nan_discrepancy = np.logical_xor(mask_nan_val_1, mask_nan_val_2)
+
+        if np.any(mask_nan_discrepancy):
+            num_pixel_nan_discrepancy = mask_nan_discrepancy.sum()
+            index_pixel_nan_discrepancy = np.where(mask_nan_discrepancy)
+            print(f'{str_indent} Found {num_pixel_nan_discrepancy} '
+                'NaN inconsistencies between input arrays. '
+                'First index of the discrepancy: '
+                f'[{index_pixel_nan_discrepancy[0][0]}]')
+            print(f'{str_indent} val_1[{index_pixel_nan_discrepancy[0][0]}] = '
+                f'{val_1[index_pixel_nan_discrepancy[0][0]]}')
+            print(f'{str_indent} val_2[{index_pixel_nan_discrepancy[0][0]}] = '
+                f'{val_2[index_pixel_nan_discrepancy[0][0]]}')
+
+            # Operations to print out further info regarding the discrapancy
+            num_nan_both = np.logical_and(mask_nan_val_1, mask_nan_val_2).sum()
+            num_nan_val_1 = np.sum(mask_nan_val_1)
+            num_nan_val_2 = np.sum(mask_nan_val_2)
+            print(f'{indent} # NaNs on val_1 only: {num_nan_val_1 - num_nan_both}')
+            print(f'{indent} # NaNs on val_2 only: {num_nan_val_2 - num_nan_both}')
+        else:
+            print(f'{str_indent} NaN discrepancy was not detected.')
+
+    # A line of space for better readability of the log
+    print('')
+
+
+def get_list_dataset_attrs_keys(hdf_obj_1: h5py.Group,
+                                key_in: str='/',
+                                list_dataset_so_far: list=None,
+                                list_attrs_so_far: list=None):
+
+    '''
+    Recursively traverse the Dataset and Attributes in the input HDF5 object.
+    Returns the list of keys for the dataset and the attributes.
+
+    NOTE:
+    In case of attributes, the path and the attribute keys are
+    separated by newline character ('\n')
+
+    Parameters:
+    ----------
+    hdf_obj_1: h5py.Group
+        HDF5 object to retrieve the dataset and the attribute list
+    key_in: str
+        path in the HDF5 object
+    list_dataset_so_far: list
+        list of the dataset keys that have found so far
+    list_attrs_so_far: list
+        list of the attribute path/keys that have found so far
+
+    Return:
+    -------
+    _ : 0 if the two HDF are identical; 1 otherwise
```

Comment:
```suggestion
    list_dataset_so_far : list
        List of datasets keys found for given HDF5 group
    list_attrs_so_far : list
        List of attributes found for given HDF5 group. Each attribute is identified by its path and key (attribute name).
```
If you commit this suggestion, please fix the number of columns to be within 80 chars.